src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -376,13 +376,55 @@ internal void FunctionEnvironmentReloadResponse(FunctionEnvironmentReloadRespons
 
             if (res.Result.IsFailure(out Exception reloadEnvironmentVariablesException))
             {
-                _workerChannelLogger.LogError(reloadEnvironmentVariablesException, "Failed to reload environment variables");
-                _reloadTask.SetException(reloadEnvironmentVariablesException);
+                if (IsEnvironmentReloadFailureFatalError(res))
+                {
+                    _workerChannelLogger.LogError(reloadEnvironmentVariablesException, "Failed to reload environment variables");
+                    _reloadTask.SetException(reloadEnvironmentVariablesException);
+                }
+                else
+                {
+                    _reloadTask.SetResult(false);
+                }
+            }
+            else
+            {
+                _reloadTask.SetResult(true);
             }
-            _reloadTask.SetResult(true);
             latencyEvent.Dispose();
         }
 
+        /// <summary>
+        /// Determines whether the environment reload failure should be considered as a fatal error or should be ignored.
+        /// </summary>
+        internal bool IsEnvironmentReloadFailureFatalError(FunctionEnvironmentReloadResponse response)
+        {
+            var workerRuntime = _environment.GetFunctionsWorkerRuntime();
+
+            // Currently we support non fatal environment reload errors for dotnet isolated worker only.
+            if (workerRuntime != RpcWorkerConstants.DotNetIsolatedLanguageWorkerName)
+            {
+                return true;
+            }
+
+            var rpcException = response.Result.Exception;
+
+            // Currently the below 2 errors are considered as non fatal.
+            if (rpcException.Message.StartsWith("Microsoft.Azure.Functions.Worker.EnvironmentReloadNotSupportedException"))
+            {
+                _workerChannelLogger.LogDebug(new EventId(422, ScriptConstants.PlaceholderMissDueToBitnessEventName),
```

Comment:
We shouldn't log this here for the same reason. 

If we want to log details, we should pass that information from the worker and have the host log what came from it.

Reply:
Working on validating an updated version...

Reply:
Pushed an update which logs the exception sent from the worker as it is. We set the "IsUserCodeException" to true so that message goes to AI logs.